src/main/java/com/passpnu/passwordmanager/util/PasswordUtil.java

```diff
@@ -0,0 +1,47 @@
+package com.passpnu.passwordmanager.util;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@Component
+public class PasswordUtil {
+
+    private final List<Pattern> patterns;
+    private final List<String> descriptions;
```

Comment:
Consider having a `Pair<String, String>` for patterns and descriptions as they should be used together

Reply:
fixed